### PR TITLE
Feature/300 respond with username when requested

### PR DESCRIPTION
**Pull request by theblockstalk**

### CODE REVIEW

File: src/sdk/helpers/requestsManager.ts [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+        const response = this.getDataSharingRequests();
+
+        if (!response) {
+            throwError('No DataSharingRequests found', SdkErrors.RequestsNotFound);
+        }
+
+        return response;
+    }
+
+    getLoginRequestWithSameOriginOrThrow(): LoginRequest {
```
Comment (sadiabbasi) [inline src/sdk/helpers/requestsManager.ts:124]: where this is used? i didn't found it

File: src/sdk/helpers/requestsManager.ts [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+        } else {
+            return [loginRequest];
+        }
+    }
+
+    getRequestsDifferentOriginOrThrow(): WalletRequest[] {
+        const loginRequest = this.getLoginRequestWithDifferentOriginOrThrow();
+        const issuer = loginRequest.getIssuer();
+
+        // TODO should maybe add origin to the DataRequest object instead of using the issuer?
```
Comment (sadiabbasi) [inline src/sdk/helpers/requestsManager.ts:166]: same as above

File: src/sdk/helpers/requestsManager.ts [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+        }
+
+        return response;
+    }
+
+    getRequestsSameOriginOrThrow(): WalletRequest[] {
+        const loginRequest = this.getLoginRequestWithSameOriginOrThrow();
+        const issuer = loginRequest.getIssuer();
+
+        // TODO should maybe add origin to the DataRequest object instead of using the issuer?
```
Comment (sadiabbasi) [inline src/sdk/helpers/requestsManager.ts:150]: don't we need to remove this as you added the origin in DataSharingRequestPayload ?

File: src/sdk/helpers/requestsManager.ts [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+        const dataSharingRequest = this.requests.find(
+            (request) => request instanceof DataSharingRequest && request.getIssuer() === issuer
```
Comment (sadiabbasi) [inline src/sdk/helpers/requestsManager.ts:152]: and here we can get the origin from request?

File: src/sdk/helpers/requestsManager.ts (hunk 1 above)
Comment (theblockstalk) [reply src/sdk/helpers/requestsManager.ts:124]: https://github.com/Tonomy-Foundation/Tonomy-App-Websites/pull/188/files#diff-37c0cb2c30e6c4ae472b0b670984a9848de8918258585cb14908f75d6810f14fR319

File: src/sdk/helpers/requestsManager.ts (hunk 2 above)
Comment (theblockstalk) [reply src/sdk/helpers/requestsManager.ts:166]: will delete

File: src/sdk/helpers/requestsManager.ts (hunk 3 above)
Comment (theblockstalk) [reply src/sdk/helpers/requestsManager.ts:150]: removed

File: src/sdk/helpers/requestsManager.ts (hunk 4 above)
Comment (theblockstalk) [reply src/sdk/helpers/requestsManager.ts:152]: yes correct

**Review by sadiabbasi — APPROVED** [review]

**Pull request merged**